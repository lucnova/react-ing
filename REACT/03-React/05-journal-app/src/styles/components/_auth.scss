.auth__main {
	align-items: center;
	background-color: $primary;
	display: flex;
	justify-content: center;
	margin: 0px;
	height: 100vh;
	width: 100vw;
}

.auth__box-container {
	background-color: white;
	box-shadow: 0px 3px lighten($color: $primary, $amount: 32);
	border-radius: 2px;
	padding: 20px;
	width: 259px;
}

.auth__title {
	color: darken($color: $primary, $amount: 32);
	margin-bottom: 64px;
}

.auth__input {
	color: $dark-gray;
	border: 0px;
	border-bottom: 1px solid $light-gray;
	font-size: 16px;
	margin-bottom: 32px;
	height: 32px;
	width: 100%;

	transition: border-bottom 0.3s ease;

	&:focus {
		border-bottom: 1px solid lighten($color: $primary, $amount: 32);
		outline: none;
	}
}

.auth__social-networks {
	display: flex;
	align-items: center;
	justify-content: center;
	flex-direction: column;
	padding-top: 20px;
	padding-bottom: 20px;
	width: 100%;
}

.auth__separator {
	margin-bottom: 32px;
	margin-top: 32px;
	color: $primary;
}

.auth__alert-error {
	background-color: red;
	border-radius: 5px;
	color: white;
	display: flex;
	justify-content: center;
	margin-bottom: 10px;
	padding: 5px;
}
