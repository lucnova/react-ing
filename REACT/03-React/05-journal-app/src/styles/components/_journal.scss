.journal__main-content {
	display: flex;
}

.journal__sidebar {
	background-color: $primary;
	color: white;
	display: flex;
	flex-direction: column;
	height: 100vh;
	padding-left: 16px;
	padding-right: 16px;
	width: 512px;
}

.journal__sidebar-navbar {
	display: flex;
	justify-content: space-between;
}

.journal__sidebar-navbar h3 {
	font-weight: lighter;
}

.journal__new-entry {
	align-items: center;
	display: flex;
	flex-direction: column;
	justify-content: center;
	margin-top: 30px;
	width: 100%;

	transition: color 0.32s ease;

	&:hover {
		color: lighten($color: $primary, $amount: 70);
	}
}

.journal__entries {
	margin-top: 30px;
	flex: 1 1 auto;
	overflow-y: scroll;
}

.journal__entry {
	background-color: white;
	border-radius: 8px;
	color: $dark-gray;
	display: flex;
	margin-bottom: 10px;
	overflow: hidden;
}

.journal__entry-picture {
	height: 75px;
	width: 75px;
}

.journal__entry-body {
	padding: 10px;
}

.journal__entry-title {
	font-size: 14px;
	font-weight: bold;
}

.journal__entry-content {
	font-size: 12px;
}

.journal__entry-date-box {
	align-items: center;
	display: flex;
	flex-direction: column;
	justify-content: center;
	padding: 5px;
}

.journal__entry-date-box span {
	font-size: 12px;
}
