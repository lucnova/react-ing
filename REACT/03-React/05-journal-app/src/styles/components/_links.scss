.link {
	color: $dark-gray;
	text-decoration: none;
	font-weight: bold;

	&:hover {
		text-decoration: underline;
	}
}
