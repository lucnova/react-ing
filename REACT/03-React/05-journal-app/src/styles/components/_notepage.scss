.notepage__main-content {
	display: flex;
	flex-direction: column;
	height: 100%;
}

.notepage__appbar {
	align-items: center;
	background-color: $primary;
	color: white;
	display: flex;
	justify-content: space-between;
	padding: 10px 20px 10px 20px;
}

.notepage__content {
	display: flex;
	flex-direction: column;
	height: 100%;
	padding: 20px;
}

.notepage__title,
.notepage__textarea {
	border: none;

	&:focus {
		outline: none;
	}
}

.notepage__title {
	color: $dark-gray;
	font-size: 25px;
	font-weight: 700;
	margin-bottom: 10px;
}

.notepage__textarea {
	border: none;
	color: $dark-gray;
	font-size: 20px;
	flex: 1 1 auto;
	resize: none;
}

.notepage__image img {
    box-shadow: 5px 5px $dark-gray;
    height: 150px;
}