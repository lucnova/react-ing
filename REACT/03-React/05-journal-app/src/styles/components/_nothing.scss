.noentry__main-content {
	align-items: center;
	background-color: lighten($color: $primary, $amount: 4);
	color: white;
	display: flex;
	flex-direction: column;
	justify-content: center;
	height: 100vh;

	p {
		text-align: center;
	}
}
