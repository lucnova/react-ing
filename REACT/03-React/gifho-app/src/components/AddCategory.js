import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState(''); // Si se deja vacio, el valor es undefined y handleChange Colapsa

	const handleInputChange = (e) => {
		setInputValue(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();

		const trimmedValue = inputValue.trim();

		if (trimmedValue.length > 2) {
			setCategories((currCategories) => {
				if (currCategories.findIndex((item) => item === trimmedValue) === -1) {
					setInputValue(''); // Limpiar Input
					return [trimmedValue, ...currCategories];
				} else {
					return currCategories;
				}
			});
		}
	};

	return (
		<>
			<p id="inputValueCheck" hidden>{inputValue}</p>

			<form onSubmit={handleSubmit}>
				<input type="text" value={inputValue} placeholder="s e a r c h" onChange={handleInputChange}></input>
				<button>o k</button>
			</form>
		</>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
