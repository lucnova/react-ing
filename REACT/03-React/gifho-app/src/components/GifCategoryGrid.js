import React /*  useState, useEffect */ from 'react'; // useEffect -> Ejecutar c칩digo de manera condicional
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { GifCategoryGridItem } from './GifCategoryGridItem';
// import { getGifs } from '../helpers/getGifs';

const GifCategoryGrid = ({ categoryName }) => {
	// const [imgs, setImgs] = useState([]);
	const {loading, data} = useFetchGifs();

	console.log(loading);

	//  - USE EFECT -
	//      Evita renderizar todo el c칩digo de nuevo para cuando hay cambios peque침os
	//      -> (callback, lista de dependencias)
	//  Si se manda vacia la lista de dependencias entonces solo se ejecuta 1 vez
	// useEffect(() => {
	// 	getGifs(categoryName)
	// 		.then(setImgs)
	// 		.catch((error) => {
	// 			console.error(error);
	// 		});
	// }, [categoryName]); // El arreglo contiene variables de las cuales useEffect se ejecutar치 denuevo

	return (
		<>
			<h3>{categoryName}</h3>
			{/* <div className="card-grid">
				{imgs.map((img) => (
					<GifCategoryGridItem key={img.id} {...img} />
				))}
			</div> */}
			{loading ? 'Loading...' : 'DONE'}
		</>
	);
};

GifCategoryGrid.propTypes = {
	categoryName: PropTypes.string.isRequired,
};

export default GifCategoryGrid;
