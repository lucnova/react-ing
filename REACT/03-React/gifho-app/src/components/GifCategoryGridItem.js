import React from 'react';
import PropTypes from 'prop-types';

export const GifCategoryGridItem = ({ url, title }) => {
	return (
		<div className="card animate__animated animate__bounceIn">
			<img src={url} alt={title}></img>
			<p>{title}</p>
		</div>
	);
};

GifCategoryGridItem.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
};
