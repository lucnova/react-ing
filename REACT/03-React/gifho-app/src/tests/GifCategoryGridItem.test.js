import { shallow } from 'enzyme';
import { GifCategoryGridItem } from '../components/GifCategoryGridItem'; // Mucho cuidado con exportaciones por defecto
// Si se cambia por defecto no se puede importar con {}

describe('TEST on GifCategoryGridItem', () => {
	const sentTitle = 'Test';
	const sentURL = 'https://localhost/algo.jpg';

	const wrapper = shallow(<GifCategoryGridItem url={sentURL} title={sentTitle} />);

	test('should display component correctly (as snapshot)', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should display same title inside his p element', () => {
		const foundText = wrapper.find('p').text().trim();

		expect(foundText).toBe(sentTitle);
	});

	test('should display a valid URL as img:src', () => {
		const foundURL = wrapper.find('img').prop('src');

		expect(foundURL).toBe(sentURL);
	});
});
