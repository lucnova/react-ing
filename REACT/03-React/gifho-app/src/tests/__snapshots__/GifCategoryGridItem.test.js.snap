// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST on GifCategoryGridItem should display component correctly (as snapshot) 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Test"
    src="https://localhost/algo.jpg"
  />
  <p>
    Test
  </p>
</div>
`;
